service/pkg/config/loader.go

```diff
@@ -0,0 +1,144 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/creasty/defaults"
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-playground/validator/v10"
+	"github.com/spf13/viper"
+)
+
+// Loader defines the interface for loading and managing configuration
+type Loader interface {
+	// Load loads the configuration into the provided struct
+	Load(cfg *Config) error
+	// Watch starts watching for configuration changes and invokes an onChange callback
+	Watch(ctx context.Context, cfg *Config, onChange func(context.Context) error) error
+	// Close closes the configuration loader
+	Close() error
+	// GetName returns the name of the configuration loader
+	GetName() string
```

Comment:
should just use the idiomatic `Name()` approach

Reply:
Updated: [`7a6490a` (#1992)](https://github.com/opentdf/platform/pull/1992/commits/7a6490a59427addab336acac6c52f48137d09901)